workers/unity/Packages/com.improbable.gdk.tools/RedirectedProcess.cs

```diff
@@ -95,6 +104,89 @@ void OnReceived(object sender, DataReceivedEventArgs args)
+        /// <param name="arguments">Parameters that will be passed to the command.</param>
+        /// <param name="redirectStdout">Redirect standard output to Debug.Log</param>
+        /// <param name="redirectStderr">Redirect standard error to Debug.LogError</param>
+        /// <returns>A task which would return the exit code and output.</returns>
+        public static async Task<RedirectedProcessResult> RunInAsync(string workingDirectory, string command, string[] arguments, bool redirectStdout = false, bool redirectStderr = false)
+        {
+            var info = new ProcessStartInfo(command, string.Join(" ", arguments))
+            {
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false,
+                WorkingDirectory = workingDirectory
+            };
+
+            return await Task.Run(() =>
+            {
+                using (var process = Process.Start(info))
+                {
+                    if (process == null)
+                    {
+                        throw new Exception(
+                            $"Failed to run {info.FileName} {info.Arguments}\nIs the .NET Core SDK installed?");
+                    }
+
+                    process.EnableRaisingEvents = true;
+
+                    var processStandardOutput = new List<string>();
+                    var processStandardError = new List<string>();
+
+                    void OnStandardOutput(object sender, DataReceivedEventArgs args)
+                    {
+                        if (!string.IsNullOrEmpty(args.Data))
+                        {
+                            if (redirectStdout)
+                            {
+                                Debug.Log(args.Data);
+                            }
+                            lock (processStandardOutput)
+                            {
+                                processStandardOutput.Add(args.Data);
+                            }
+                        }
+                    }
+
+                    void OnStandardError(object sender, DataReceivedEventArgs args)
+                    {
+                        if (!string.IsNullOrEmpty(args.Data))
+                        {
+                            if (redirectStderr)
+                            {
+                                Debug.LogError(args.Data);
+                            }
+                            lock (processStandardOutput)
+                            {
+                                processStandardError.Add(args.Data);
+                            }
+                        }
+                    }
+
+                    process.OutputDataReceived += OnStandardOutput;
+                    process.ErrorDataReceived += OnStandardError;
+
+                    process.BeginOutputReadLine();
+                    process.BeginErrorReadLine();
+
+                    process.WaitForExit();
```

Comment:
I am not 100% familiar with async handling in c#, but I am curious if WaitForExit would actually make this task async or just block the thread.
Should we use something like described in https://stackoverflow.com/questions/470256/process-waitforexit-asynchronously instead?

Reply:
This is already in an async context (see Task.Run above), so it wouldn't block the main thread.